### PR TITLE
Ccs functionality

### PR DESCRIPTION
**Pull request by sam-nguyen7**
### What type of PR is this? 
feature

### What this PR does / why we need it:
Adds CCS functionality
### Which issue(s) this PR fixes(can be a reference to existing issue or jira/bz):

_Fixes #_

### Pre-checks:
- [ ] manually tested latest changes against crc/k8s
- [ ] run the `make coverage` command to generate new calculated coverage

### CODE REVIEW

**Comment by sam-nguyen7** [comment]
/hold 
for testing

File: pkg/controller/projectreference/projectreference_adapter.go [hunk 1]
```diff
@@ -281,12 +298,14 @@ func (r *ReferenceAdapter) EnsureFinalizerDeleted() error {
...
 // EnsureProjectCleanedUp deletes the project, the secret and the finalizer if they still exist
 func (r *ReferenceAdapter) EnsureProjectCleanedUp() error {
-	err := r.deleteProject()
-	if err != nil {
-		return err
+	if !r.isCCS() {
+		err := r.deleteProject()
+		if err != nil {
+			return err
+		}
```
Comment (lisa) [inline pkg/controller/projectreference/projectreference_adapter.go:305]: OSD-5048

**Comment by NautiluX** [comment]
/lgtm
/approve
ready to merge from reading the code,
remove hold once you're done with manual testing

**Pull request merged**